src/lib/logger/impl.ml

```diff
@@ -357,7 +357,22 @@ let raw ({ id; _ } as t) msg =
   if t.null then ()
   else if Message.check_invariants msg then
     Consumer_registry.broadcast_log_message ~id msg
-  else failwithf "invalid log call \"%s\"" (String.escaped msg.message) ()
+  else
+    let msg' =
+      Message.
```

Comment:
You should probably do `{msg with level = Error; message = ...}` here instead. This will preserve the any global or local metadata in the log message, which could be useful for debugging.

Reply:
Yeah... I thought about that but wanted to make it simple to make sure say nothing else from logging logic triggers the fail.

Like.. some information is dropped, but I don't see it as an issue, as this line is a safeguard for something that shall not generally happen